src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -991,6 +1032,11 @@ private static void GetSimplyNamedTypeFromAssembly(Assembly assm, string typeNam
             catch (FileNotFoundException) { }
             catch (FileLoadException) { }
             catch (BadImageFormatException) { }
+
+            if((object)type == null)
```

Comment:
Casting type to object before comparing to null isn't necessary. The == operator for MemberInfo ensures that null checks occur without invoking any members on potentially 3-rd party types (assuming such a thing could even make it to this point.)